src/Covenant/ASG.hs

```diff
@@ -208,14 +241,15 @@ lam Scope f = do
 --
 -- @since 1.0.0
 letBind ::
-  forall (n :: Natural) (m :: Natural).
-  Scope n m ->
+  forall (args :: Natural) (binds :: Natural) (m :: Type -> Type).
+  (MonadHashCons Id ASGNode m) =>
+  Scope args binds ->
   Ref ->
-  (Scope n (m + 1) -> ASGBuilder Ref) ->
-  ASGBuilder Id
+  (Scope args (binds + 1) -> m Ref) ->
+  m Id
```

Comment:
I like the type variables names refactor :+1:

Reply:
I should have done this from the start.